### ISSUE TITLE
About the input shape of predictor.predict

### ISSUE THREAD

**Issue by meyurtsever** [maxmekiska/Imbrium#3]
Hi there. Great project. I gave it a shot and encountered an issue.

I've trained the model using `PureMulti`, `create_lstm` and `fit_model` methods.

This is the X_input shape from the predictor (96 is window size): `(26107, 96, 7)`

And I want to get the predictions from the separate dataset. I've transformed my test dataset using `data_prep_multi` and `multistep_prep_standard` methods. After that, the test dataset shape is `(8575, 96, 7)` but still getting dimension/shape errors.

So, what is the proper format of data while passing it to predictor.predict?

**Comment by maxmekiska** [comment]
Hi @meyurtsever,

Thank you very much for using imbrium and raising this to my attention. I apologize for the inconvenience caused by this error. 

Could you please verify that you have used the PureMulti class in the following way:

```python
from imbrium.predictors.multivarpure import *

predictor = PureMulti(steps_past =  5, steps_future = 10, data = data, features = ['target_variable', 'HouseAge', 'AveRooms', 'AveBedrms'])
```
Please observe in the above how I have set the features to be considered during training. The first variable: `target_variable`  is the target variable which you want to make predictions for. The rest `HouseAge`, `AveRooms`, and `AveBedrms` are the features that will be used to predict the `target_variable`.

```python
predictor.create_lstm()

predictor.fit_model(epochs = 20)

predictor.show_performance()
```
After a successful training process, you can now use your predictor to make predictions on data with a similar shape. Please note that the data (dataframe) you will have to pass in, must NOT contain the target variable `target_variable`. It should only contain the features:  `HouseAge`, `AveRooms`, and `AveBedrms`. 

For example:

use new data and make a prediction based on a past window of 5 days, 10 days into the future:

```python
predictor.predict(data[['HouseAge', 'AveRooms', 'AveBedrms']].tail(5))
```

I hope this helps.

**Comment by meyurtsever** [comment]
Thank you @maxmekiska for your quick and explanatory answer.  I forgot to remove the target variable in the test dataset. After fixing that, I can now get predictions.

In my case, I use 96 window size (hourly data) for prediction. But no matter the input window of x days, the model always produces (96, 1) predictions. So, how to evaluate predicted values with the test dataset? How can I spot the ground truth data to compare with predicted results?

**Comment by maxmekiska** [comment]
@meyurtsever I'm glad this solved your problem. I will work to improve the documentation for imbrium in future releases.

I believe you are asking about back-testing?

The current behavior of the model is expected. If you want to do back-testing using your trained predictor, I would recommend writing a custom loop. This loop should call the predictor with the window size defined during training, capture the predictor's forecasts, and compare them with the true data.

Please note that during training you can already get metrics on the loss of your predictor:

```python
predictor1.fit_model(epochs = 200, show_progress= 1, validation_split = 0.20)  # 20% used of training data for validation
predictor1.show_performance() # shows loss chart
```

I hope this helps. I will consider enhancing the show_performance() method in future releases to display other metrics in addition to the loss function.

**Comment by meyurtsever** [comment]
Hey, @maxmekiska. Thanks for your reply. Yes, I can observe the loss while training. I also wanted to test it with unseen data from the model's end. As you suggested, writing a custom loop and comparing the results will be the best for the testing data apart from training.

I thank you again for developing such a rich and easy-to-implement framework. Much appreciated!
